Automattic/wp-calypso
# Fix: locale redirect on pricing page

#### Changes proposed in this Pull Request

Currently, the checkout page on Calypso doesn't take language into account. It means when redirected from cloud.jetpack.com and jetpack.com to checkout, not logged in, language will be English by default. We are adding to two URLs: `/checkout/jetpack/:productSlug` and `/checkout/jetpack/:siteSlug/:productSlug` lang param in the end to handle language.
In this PR we are also passing it from cloud.jetpack.com side.

Note: jetpack.com -> wordpress.com/checkout... will be handled separately.

#### Testing instructions
To test locally:
- Checkout the branch locally.
- `yarn install`
- `yarn start-jetpack-cloud` 
- `PORT=443 PROTOCOL=https yarn start`
- Add next line to hosts file (or using gasmask) `127.0.0.1 wordpress.com` and 'flush dns'

- navigate to http://jetpack.cloud.localhost:3000/pricing , then try to purchase a product. You should be redirected correctly to `wordpress.com/checkout...` with default language.
- navigate to  http://jetpack.cloud.localhost:3000/fr/pricing or change the language manually to not default. You should be redirected to a proper checkout page with the language selected previously.
- These steps should also work with site selected when purchase on `checkout/jetpack`. On cloud jetpack when `canDoUnlinkedCheckout` is set to true.
- Need also to check that all links to checkout from cloud.jetpack.com and jetpack.com to wordpress.com/checkout are working properly.

Related to #
1164141197617539-as-1201508699348275/f

Discussion:
p9oQ9f-10K-p2
